packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -425,13 +426,15 @@ class _HeroFlightManifest {
   final CreateRectTween createRectTween;
   final HeroFlightShuttleBuilder shuttleBuilder;
   final bool isUserGestureTransition;
+  final bool isDiverted;
 
   Object get tag => fromHero.widget.tag;
 
   Animation<double> get animation {
     return CurvedAnimation(
       parent: (type == HeroFlightDirection.push) ? toRoute.animation : fromRoute.animation,
       curve: Curves.fastOutSlowIn,
+      reverseCurve: isDiverted ? null : Curves.fastOutSlowIn.flipped,
```

Comment:
Does this mean diverted heroes use `Curves.fastOutSlowIn` as a reverseCurve? When does this actually happen?

Reply:
Yes. A diverted hero is a hero that was in-flight and as it was flying got a new destination assigned to it. This can happen when you push a route and as the route is transitioning in you pop it again. The hero would start flying to the newly pushed route, but then as that one is popped fly back to where it came from. Usually, the pop should use the reverse curve, but in this particular case we do not want to switch curves as that would make the hero jump from the position it had on `curve` to the one it had on `reverseCurve`.